libevm/VMOpt.cpp

```diff
@@ -119,9 +121,13 @@ void VM::optimize()
 		}
 #endif
 	}
-	
-	copyCode(pc - nBytes + 33);
 
+	if (pc > nBytes)
+	{
+		m_codeSpace.resize(m_codeSpace.size() + pc - nBytes);
```

Comment:
I think it is better to create a helper function (which is also used by `copyCode`) that does the following:
```
m_codeSpace.resize(size);
m_code = m_codeSpace.data()
```